### PR TITLE
feat: add jsdoc for diff-match-patch

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/google/diff-match-patch/blob/62f2e689f498f9c92dbc588c58750addec9b1654/javascript/diff_match_patch_uncompressed.js>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/diff-match-patch/index.d.ts [hunk 1]
```diff
@@ -1,63 +1,288 @@
...
+export type Diff = readonly [
+    /**
+     * Operation, one of: DIFF_DELETE, DIFF_INSERT, DIFF_EQUAL
+     */
+    operation: Operation,
+    /**
+     * Text to be deleted, inserted, or retained.
+     */
+    text: string
+];
```
Comment (peterblazejewicz) [inline types/diff-match-patch/index.d.ts:29]: The enum notion does not exists in original code, can this be done just using aliases to provide proper types?
```ts
export const DIFF_DELETE: -1;
export const DIFF_INSERT: 1;
export const DIFF_EQUAL: 0;

export type Diff = readonly [
    /**
     * Operation, one of: DIFF_DELETE, DIFF_INSERT, DIFF_EQUAL
     */
    operation: typeof DIFF_DELETE | typeof DIFF_INSERT | typeof DIFF_EQUAL,
    /**
     * Text to be deleted, inserted, or retained.
     */
    text: string,
];
```

File: types/diff-match-patch/tslint.json [hunk 2]
```diff
@@ -13,6 +13,8 @@
                 ],
                 "mode": "code"
             }
-        ]
+        ],
+        "no-const-enum": false,
```
Comment (peterblazejewicz) [inline types/diff-match-patch/tslint.json:17]: the union options would allow you to disregards enums

File: types/diff-match-patch/tslint.json [hunk 3]
```diff
@@ -13,6 +13,8 @@
...
+        "unified-signatures": false
```
Comment (peterblazejewicz) [inline types/diff-match-patch/tslint.json:18]: Please use this inline (`// tslint:disable:....`) as per README, unless there is a large number of affected lines

**Review by amcasey — CHANGES_REQUESTED** [review]
Please address the feedback from @peterblazejewicz

File: types/diff-match-patch/index.d.ts (hunk 1 above)
Comment (xiaoxiangmoe) [reply types/diff-match-patch/index.d.ts:29]: But if we use `typeof DIFF_DELETE | typeof DIFF_INSERT | typeof DIFF_EQUAL` it will become `-1 | 0 | 1`

**Pull request closed without merging**